utils/crud/backingStore.go

```diff
@@ -0,0 +1,119 @@
+package crud
+
+var _ Store = &BackingStore{}
+
+// BackingStore wraps another store that may have Connect/Close methods that
+// need to be called.
+// - Connect is called before a method when the connection is closed.
+// - Close is called after each method when AutoClose is true (default).
+type BackingStore struct {
+	AutoClose    bool
+	closed       bool
+	backingStore Store
+}
+
+func NewBackingStore(store Store) *BackingStore {
+	return &BackingStore{
+		AutoClose:    true,
+		closed:       true,
```

Comment:
nit: What about renaming `closed` to `opened` ? I think it fits better here in the "constructor":
```go
opened: false,
```
And also below in the `Connect` function, I think we should avoid double negations:
```go
if s.opened{
    return nil
}
```
WDYT?

Reply:
I was thinking about that, works better too because the default is false without having to set it in a constructor. I'll update it.